pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,402 @@
+                    # connect axis main view changes to view
+                    cvb.state["linkedViews"][cvb.YAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigYRangeChanged"] = axis_view.sigYRangeChanged.connect(
+                        cvb.linkedYChanged)
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigResized"] = axis_view.sigResized.connect(
+                        cvb.linkedYChanged)
+                    # disable autorange on manual movements
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigYRangeChangedManually"] = axis_view.sigYRangeChangedManually.connect(
+                        lambda mask: self.disableAxisAutoRange(axis_name))
+            self._signalConnectionsByChart[chart.name]["cvb.sigStateChanged"] = cvb.sigStateChanged.emit(
+                cvb)
+        # resize plotitem according to the master one
+        # resizing it's view doesn't work for some reason
+        self._signalConnectionsByChart[chart.name]["self.vb.sigResized"] = self.vb.sigResized.connect(
+            lambda vb: chart.plotItem.setGeometry(vb.sceneBoundingRect()))
+        # fix prepareForPaint by outofculture
+        self._signalConnectionsByChart[chart.name]["scene.sigPrepareForPaint"] = scene.sigPrepareForPaint.connect(
+            cvb.prepareForPaint)
+        if cvb is not tvb:
+            # FROM "https://github.com/pyqtgraph/pyqtgraph/pull/2010" by herodotus77
+            # propagate mouse actions to charts "hidden" behind
+            self._signalConnectionsByChart[chart.name]["tvb.sigMouseDragged"] = tvb.sigMouseDragged.connect(
+                cvb.mouseDragEvent)
+            self._signalConnectionsByChart[chart.name]["tvb.sigMouseWheel"] = tvb.sigMouseWheel.connect(
+                cvb.wheelEvent)
+            self._signalConnectionsByChart[chart.name]["tvb.sigHistoryChanged"] = tvb.sigHistoryChanged.connect(
+                cvb.scaleHistory)
+
+    def disconnect_all(self, chart):
+        """Disconnects all signals related to this widget for the given chart."""
+        for conn_name, conn in self._signalConnectionsByChart[chart.name].items():
+            if conn is not None:
+                try:
+                    self.disconnect(conn)
+                except (TypeError, RuntimeError):
+                    pass
+                self._signalConnectionsByChart[chart.name][conn_name] = None
+
+    def clean(self):
+        """Clears all charts' contents."""
+        # CLEAR PLOTS
+        for p in self.charts.values():
+            p.clear()
+        self.update()
+
+    def getPlotItem(self, name=None):
+        """Get the PlotItem associated to the chart of given name.
```

Comment:
The following docstring won't emit warnings

```python
        """Get the PlotItem associated to the chart of given name.

        Parameters
        ----------
        name : str or None
            The name of the chart to select.
            If None the default one will be selected.
        Returns
        -------
            PlotItem
                The PlotItem associated to the selected chart.
        """
```

Reply:
done